include/valijson/validation_results.hpp

```diff
@@ -51,6 +51,36 @@ class ValidationResults
     };
 
     /**
+     * @brief  Iterator for all validation errors.
+     *
+     * This struct is used to get a reference to an iterator for errors.
+     */
+    struct ErrorRange
+    {
+        typedef std::deque<Error>::const_iterator ErrorIterator;
+        ErrorRange(const std::deque<Error> &errors) : m_errors(errors) {};
+
+        ErrorIterator begin() {
+            return m_errors.begin();
+        }
+
+        ErrorIterator end() {
+            return m_errors.end();
+        }
+
+    private:
+        const std::deque<Error> &m_errors;
```

Comment:
While we're looking at this class, the destructor does not need to be `virtual` since this class is never used polymorphically and does not have any `virtual` member functions.